geo-types/Cargo.toml

```diff
@@ -19,12 +19,12 @@ use-rstar_0_8 = ["rstar_0_8", "approx"]
 use-rstar_0_9 = ["rstar_0_9", "approx"]
 
 [dependencies]
-approx = { version = "0.4.0", optional = true }
+approx = { version = "0.5.1", optional = true }
```

Comment:
> This would constitute a breaking change for geo-types, right?

Can I just understand the thought process here?

"Approx 0.5.1"

So one of our sub packages is making a minor number upgrade ... 

Under the semi version rules as I understand them an increase in the minor number implies any features that have been added do not affect the API. So they are either modification to internal algorithms, or the addition of extra functions.

As I read it ONLY if approx had bumped them major version number - then would that be a breaking change?
Maybe I am wrong?

I am just trying to understand the rules of the road here... because as I said in #849 
My interest is more than fixing approx. it is keeping the technical debt of outdated packages manageable.

I am happy to do the leg work, and maybe  add other packages to the PR. Or submit a series of PR taking this or that into consideration.

I know than means lots of consultation and it is going to be a series painful steps keeping the thing updated. 
but in my experience the alternative is the project becomes locked into an un-upgrade-able state 
as the active contributors needs diverge..

What I mean is say  group A needs the version to be on these numbers but  group B need a alternate set of sub-dependencies.

Reply:
> Under the semi version rules as I understand them an increase in the minor number implies any features that have been added do not affect the API. So they are either modification to internal algorithms, or the addition of extra functions.

I think you are right - but only when talking about packages that are *post* 1.0.0 release. 

Packages that are pre-1.0.0 (`0.y.z`) technically can break anything and everything with every release, but in practice what I see is more conservative: that minor versions (bumping the `y`)  can (and typically do) add breaking changes while bumps to the patch (`z`) are semver safe.

More to the point: If we bump approx to 0.5, while other libraries that depend on us are still using approx 0.4, then cargo will no longer resolve a compatible version between the crates.

A third option occurs to me, since we're not depending on any of the changes introduced in 0.5, could we loosen up and accept a range of approx versions?


Something like:
```
approx = ">= 0.4.0, < 0.6.0"
```

I think that'd allow people to use the latest while not forcing them to upgrade, right?